static/components/TaskDetails.jsx

```diff
@@ -124,40 +136,124 @@ var TaskDetails = React.createClass({
     });
   },
+
+  splitLogData(data){
+    var text       = data.chunks;
+    var objLength  = data.chunks.length;
+
+    for(var i = 0; i < objLength; i++){
+      if(data.chunks[i] != undefined){
+        this.setState({
+          lines: [...this.state.lines, text[i].text.split(/\n/)],
+          timeStamp: [...this.state.timeStamp, data.chunks[i].date],
+        })
+      }
+    }
+  },
+
+  highLightDiv(div){
+    div.addEventListener('click', (e) => {
+      var div                   = document.getElementById(e.target.id);
+      var highlightedlines      = document.getElementsByClassName('line highLighted');
+      var lines                 = document.getElementsByClassName('line');
+
+      this.stopRefresh(e, e.target.href);
+
+      if(div.className == 'line'){
+        div.className = 'line highLighted';
+
+      } else if(div.className == 'line highLighted'){
+        div.className = 'line';
+        this.stopRefresh(e, e.target.href);
+      }
+
+      for(var l = 0 ; l < highlightedlines.length; l++){
+        if(highlightedlines[l].className == 'line highLighted'){
+          highlightedlines[l].className = 'line';
+          div.className = 'line highLighted';
+        }else{
+          div.className = 'line';
+        }
+      }
+    });
+  },
+
   updateBuildLog(data) {
     // add each additional new line
+    this.splitLogData(data);
     var frag       = document.createDocumentFragment();
-    var text       = data.chunks
-    var objLength  = data.chunks.length
+    var timeid     = 0;
+    var index      = 0;
+    var lineItem   = this.state.lines;
+    var hash       = window.location.hash;
 
-    for(var i = 0; i < objLength; i++){
-      var timer    = new Date(data.chunks[i].date)
-      var timeMil  = timer.getTime()
+    for(var j = this.state.index; j < lineItem.length; j++){
+      for(var k = 0; k < lineItem[j].length; k++){
 
-      //Multiple by 60000 to convert offset to milliseconds
-      var offset   = timer.getTimezoneOffset() * 60000
-      var timezone = timeMil - offset
-      var newDate  = new Date(timezone)
+        let div         = document.createElement('div');
+        let time        = document.createElement('a');
+        let idIncrement = this.state.id++;
 
-      var div  = document.createElement('div');
-      var time = document.createElement('div');
+        div.className  = 'line';
+        time.className = 'time';
+        div.id         = 'L' + idIncrement;
 
-      div.className  = 'line';
-      time.className = 'time';
+        let {environment, number} = this.state.task;
+        let {name}                = this.state.task.app;
 
-      div.innerHTML  = ansi_up.ansi_to_html(data.chunks[i].text)
-      time.innerHTML = moment(newDate).parseZone().format("h:mm:ss a")
+        time.href = `/deploys/${name}/${environment}/${number}#${div.id}`;
+        time.id   = div.id;
 
-      frag.appendChild(time)
-      frag.appendChild(div)
-    }
+        this.highLightDiv(time);
```

Comment:
You're still creating a new event listener for every line element. Compare this vs a single listener on a parent container (say `this.refs.log`).  The downside is a bit more complexity having the event bubble up to the container and making sure the element you want was clicked. Maybe just compare memory usage/perf on this vs no event listener.